### ISSUE TITLE
get_lr depends on iteration but not step

### ISSUE THREAD

**Issue by Lavenderjiang** [Lightning-AI/lit-gpt#926]
[get_lr](https://github.com/Lightning-AI/lit-gpt/blob/01ae9d37ef464b90de2aa6d00df3c985c65cbc56/pretrain/redpajama.py#L308) depends on iteration (one forward and backward pass) but not step (one optimizer step). 

This could be a problem when we have gradient accumulation (where 1 step= n iterations). Here we would change the learning rate n times before calling one optimizer.step()

Is there a particular reason why we chose to make get_lr depends on iter?
